validate packet decap based on number of IP layers until UDP layer

dracazacula opened the pull request:
Decoding gopackets might lead to errors for specific [UDP ports](https://github.com/google/gopacket/blob/master/layers/ports.go#L114) by trying to decode the UDP payload into other protocol layers, which leads to false negatives when checking if captured packets have been decapped.

The validation function changed to counting the number of IP layers strictly before the UDP layers, since the generated traffic is UDP with no specific meaning in the payload. 

Trying to stop the decoding up until UDP layer is not possible.

singhavnish2516 commented:
/fptest virtual

singhavnish2516 commented on feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go:
```diff
@@ -1763,16 +1774,8 @@ func validateTrafficDecap(t *testing.T, captureFile *os.File, expectedInHdrDscp
-				t.Errorf("validateTrafficDecap: decap success rate for IPv4 packets is less than 99.0%%")
-			}
-		}
+		t.Errorf("validateTrafficDecap: a number of %v packets have not been decapped out of %v IPv4 packets captured",
```
`a number of` is not required

singhavnish2516 commented on feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go:
```diff
@@ -1717,18 +1726,20 @@ func validateTrafficDecap(t *testing.T, captureFile *os.File, expectedInHdrDscp
+				ipv6InnerLayer := innerPacket.Layer(layers.LayerTypeIPv6)
+				if ipInnerLayer != nil {
+					testStats.IPv4NotDecappedPackets++
+					log.Warningf("validateTrafficDecap: packets are not decapped, Inner IP header is not removed")
```
nit: inner instead of Inner

dracazacula replied:
sure, done!

dracazacula commented on feature/gribi/otg_tests/vrf_policy_driven_te/vrf_policy_driven_te_test.go:
```diff
@@ -1783,16 +1786,8 @@ func validateTrafficDecap(t *testing.T, captureFile *os.File, expectedInHdrDscp
-				t.Errorf("validateTrafficDecap: decap success rate for IPv6 packets is less than 99.0%%")
-			}
-		}
+		t.Logf("validateTrafficDecap: a number of %v packets have not been decapped out of %v IPv6 packets captured",
```
done!

dracazacula commented:
> > Decoding gopackets might lead to errors for specific [UDP ports](https://github.com/google/gopacket/blob/master/layers/ports.go#L114) by trying to decode the UDP payload into other protocol layers, which leads to false negatives when checking if captured packets have been decapped.
> > I changed the validation function to count the number of IP layers strictly before the UDP layers, since the traffic we generate is UDP with no specific meaning in the payload.
> > Trying to stop the decoding up until UDP layer is not possible.
> 
> Prefer writing description in third person

Makes sense, changed to third person.
